homeassistant/components/device_tracker/cppm_tracker.py

```diff
@@ -0,0 +1,86 @@
+"""
+Support for ClearPass Policy Manager.
+
+Allows tracking devices with CPPM.
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, DeviceScanner, DOMAIN
+)
+from homeassistant.const import (
+    CONF_HOST, CONF_API_KEY
+)
+
+REQUIREMENTS = ['clearpasspy==1.0.2']
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+CLIENT_ID = 'client_id'
+
+GRANT_TYPE = 'client_credentials'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CLIENT_ID): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_scanner(hass, config):
+    """Initialize Scanner."""
+    from clearpasspy import ClearPass
+    data = {
+        'server': config[DOMAIN][CONF_HOST],
+        'grant_type': GRANT_TYPE,
+        'secret': config[DOMAIN][CONF_API_KEY],
+        'client': config[DOMAIN][CLIENT_ID]
+    }
+    cppm = ClearPass(data)
+    if cppm.access_token is None:
+        return None
+    _LOGGER.debug("Successfully received Access Token.")
```

Comment:
We don't end logging messages with period.

Reply:
I'm starting to think you're picking on me. :)

Reply:
Actually, it's because I got dinged so many times by pylint for NOT using periods in the doc-strings.